models/classes/accessControl/AbstractContext.php

```diff
@@ -6,6 +6,9 @@
 
 use InvalidArgumentException;
 
+/**
+ * @deprecated Use \oat\tao\model\Context\AbstractContext
+ */
 abstract class AbstractContext
```

Comment:
You can erase all the content and extend the proper abstraction, so you do not keep the same logic in 2 places.

```suggestion
abstract class AbstractContext extends \oat\tao\model\Context\AbstractContext
```